dbt/compilation.py

```diff
@@ -352,8 +352,16 @@ def compile_nodes(self, linker, nodes, macro_generator):
             # wrapping
             if injected_node.get('resource_type') in [NodeType.Test,
                                                       NodeType.Analysis]:
-                # don't wrap tests or analyses.
-                injected_node['wrapped_sql'] = injected_node['injected_sql']
+                if 'data' in injected_node['tags']:
+                    # data tests get wrapped in count(*)
+                    injected_node['wrapped_sql'] = (
+                        "select count(*) from (\n{test_sql}\n) sbq").format(
```

Comment:
this is ok for now, but i'm not crazy about inlining the sql here, and using a tag to decide whether or not to wrap. this means analyses with the data tag would get wrapped, which is a little odd.